src/network/networkd-link.c

```diff
@@ -1470,17 +1470,20 @@ static int link_initialized(Link *link, sd_device *device) {
         assert(link);
         assert(device);
 
-        if (link->state != LINK_STATE_PENDING)
-                return 0;
+        /* Always replace with the new sd_device object. As the sysname (and possibly other properties
+         * or sysattrs) may be outdated. */
+        sd_device_unref(link->sd_device);
+        link->sd_device = sd_device_ref(device);
```

Comment:
doesn't matter, but for replacements like this i'd alway ref first, and unref second, just in case the object passed in is actually the one already stored in there. because if you unref first then it might be destroyed already, and then you ref invalidated data...

Reply:
Interesting. Maybe, we should introduce something similar to `free_and_replace()` but with `_ref()` and `_unref()`.